examples/get_started/common_sql_functions.py

```diff
@@ -25,19 +25,20 @@ def num_chars_udf(name):
 )
 
 (
-    ds.mutate(stem=path.file_stem(C("file.name")), ext=path.file_ext(C("file.name")))
+    dc.mutate(stem=path.file_stem(C("file.name")), ext=path.file_ext(C("file.name")))
     .select("file.name", "stem", "ext")
     .show(5)
 )
 
+chain = dc.mutate(
+    a=array.length(string.split(C("file.parent"), literal("/"))),
+    b=array.length(string.split(C("file.name"), literal("0"))),
+)
+
 (
-    ds.mutate(
-        a=array.length(string.split(C("file.parent"), literal("/"))),
-        b=array.length(string.split(C("file.name"), literal("0"))),
-    )
-    .mutate(
-        greatest=greatest(C("a"), C("b")),
-        least=least(C("a"), C("b")),
+    chain.mutate(
+        greatest=greatest(chain.column("a"), chain.column("b")),
```

Comment:
[F] This is a patch for https://github.com/iterative/datachain/pull/226

Reply:
alternative patch is 

```python
(
    dc.mutate(
        a=array.length(string.split(C("file.parent"), literal("/"))),
        b=array.length(string.split(C("file.name"), literal("0"))),
    )
    .mutate(
        greatest=greatest(C("a", Int64), C("b", Int64)),
        least=least(C("a", Int64), C("b", Int64)),
    )
    .select("a", "b", "greatest", "least")
    .show(10)
)
```

Reply:
and the last is to give `greatest`/`least` a `type=Float()` property and change it from inheriting `ReturnTypeFromArgs` to `GenericFunction`

```python
class greatest(GenericFunction):  # noqa: N801
    inherit_cache = True

    type = Float()


class least(GenericFunction):  # noqa: N801
    inherit_cache = True

    type = Float()

```